src/Bicep.Core/TypeSystem/DeployTimeConstantContainerVisitor.cs

```diff
@@ -32,7 +32,19 @@ public static IEnumerable<SyntaxBase> CollectDeployTimeConstantContainers(Semant
 
         public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax syntax)
         {
-            if (!syntax.IsExistingResource())
+            if (syntax.IsExistingResource())
+            {
+                var body = syntax.TryGetBody();
+
+                if (body is not null)
+                {
+                    foreach (var nestedResource in body.Resources)
+                    {
+                        this.Visit(nestedResource);
+                    }
+                }
```

Comment:
Should the `if (!syntax.IsExistingResource())` condition be removed instead? Nested resource visitation would happen already as part of `AstVisitor::VisitResourceDeclarationSyntax`?

Reply:
No, it's an edge case we need to handle. Basically, DTC validation for existing resource properties should be skipped. For example, the following is valid for an existing resource:

```
module foo './foo.bicep' = { ... }

resource bar '...' = {
  name: foo.outputs.xyz // Can be a non-DTC value
}
```

I'll add a comment to explain it.